common/src/main/java/org/apache/rocketmq/common/KeyBuilder.java

```diff
@@ -31,6 +31,13 @@ public static String parseNormalTopic(String topic, String cid) {
         }
     }
 
+    public static boolean isPopRetryTopicOfTopic(String retryTopic, String topic) {
+        if (!retryTopic.startsWith(MixAll.RETRY_GROUP_TOPIC_PREFIX)) {
+            return false;
+        }
+        return retryTopic.endsWith("_" + topic);
```

Comment:
If there is a group called GroupA subscribe a topic called TopicA, there will have a pop retry topic called %RETRY%GroupA_TopicA.
But If there is another group called GroupB subscribe a topic called Another_TopicA, there will have a pop retry topic called %RETRY%GroupB_Another_TopicA.
Therefore, when deleting TopicA, other pop retry topic will be deleted by mistake.

Reply:
Indeed, I'll check the consumer group names before deleting pop retry topics.